python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -968,10 +967,16 @@ def has_loaded(self, obj):
         Args:
             obj (GangaObject): Object which we want to look for in this repo
         """
+        obj_id = id(obj)
+        self.lock_transaction(obj_id, 'has_loaded')
```

Comment:
good point. Although at least this protects against this object being removed while we're checking if it's in memory... We desperately need some form of get/set lock system in the Registry.